### ISSUE TITLE
Add true PiP for voip/video calls and possibly even Jitsi

### ISSUE THREAD

**Issue by ara4n** [vector-im/element-web#15887]
HTML gives you true PiP these days - as per https://www.w3.org/TR/picture-in-picture/ and https://caniuse.com/picture-in-picture.  We should use it.

**Comment by t3chguy** [comment]
iirc this only works for `<video />` elements so would mean you wouldn't have any buttons and/or your own preview.

**Comment by ara4n** [comment]
yup.

**Comment by ara4n** [comment]
Alternatively, we could use proper native OS PIP in Electron, like Jitsi does.  This is the main reason why I don't use Element for video calls - Jitsi's PIP is invaluable.

**Comment by t3chguy** [comment]
Yes, though using Jitsi Electron's PIP has the same issue as screensharing in Electron with the sandboxing/security preventing us from easily solving it as we can't attach things onto a cross-origin iframe like Jitsi expects us to do.

**Comment by ara4n** [comment]
presumably Electron lets you override the cross-origin iframe sandboxing if you try hard enough? or is this https://github.com/electron/electron/issues/23664?

**Comment by ara4n** [comment]
stupid question: can we add vector:// as a valid origin on our jitsi?

**Comment by t3chguy** [comment]
> presumably Electron lets you override the cross-origin iframe sandboxing if you try hard enough?

Not in any granular way last I checked, it requires turning off a bunch of security in the app and not just scoped to a given iframe.

https://github.com/vector-im/element-web/issues/4880 has a lot more discovery

**Comment by SimonBrandner** [comment]
We should also consider adding a screen-sharing floating bar (see https://www.figma.com/file/V6m2z0oAtUV1l8MdyIrAep/VoIP?node-id=4999%3A1913):
![Screenshot_20210730_095710](https://user-images.githubusercontent.com/25768714/127620937-6e2e257d-6b80-4baf-9dd2-852ecb0d3dfd.png)
